Updated Readme.md

bgelens opened the pull request:
* Fixes: #112
* Added missing or corrected DataType and Dsc attribute info
* Added appveyor Dev badge
* Added missing parameter info
* Fixed / normalized examples
* Added Opt In for Markdown common tests

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/powershell/xhyper-v/113)
<!-- Reviewable:end -->

bgelens commented:
@johlju or @PlagueHO Can anyone of you sign this off? Thanks!

johlju commented:
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 32 unresolved discussions.

---

*[README.md, line 4 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1rGjRMqhBpfWbBCsq:-Ko1rGjSc76oqeW93tyP:bikcc21) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L4)):*
> ```Markdown
> 
> The **xHyper-V** DSC module configures and manages a Hyper-V host using the
>  **xVhd**, **xVMHyperV**, **xVMSwitch**, **xVhdFile**, **xVMDvdDrive**,
> ```

Do we need to list each resource here? Seems it could be missed when new resources are added. Could we just say something like "using Desired State Configuration." instead?
Or xSQLServer has the text 'The xSQLServer module contains DSC resources for deployment and configuration of SQL Server.'. So maybe something along those lines?

But if you want to keep it as-is, then that looks okay to me.

---

*[README.md, line 31 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1sneoRY9-xOxeW-VC:-Ko1snepIyrACAJtTp3C:bexk9ql) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L31)):*
> ```Markdown
> ## Resources
> 
> * [**xVhd**](#xvhd) manages VHDs in a Hyper-V host.
> ```

Could you order these resources in alphabetical order?

---

*[README.md, line 31 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1tHRd5k3qEOOsVZtP:-Ko1tHRevm2tkSLzMsew:bvzwdkj) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L31)):*
> ```Markdown
> ## Resources
> 
> * [**xVhd**](#xvhd) manages VHDs in a Hyper-V host.
> ```

The resource and the resource folder is named xVHD (upper 'VHD'). This should reflect that. Or the resource should change the name to 'xVhd'.

---

*[README.md, line 47 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1tgOSNlQt_erSvyTR:-Ko1tgOSNlQt_erSvyTS:b-amhpbo) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L47)):*
> ```Markdown
>  a Hyper-V virtual machine or to the management OS.
> 
> ### xVhd
> ```

This resource has read parameters that should be listed here as well.

Example:
https://github.com/PowerShell/PSDscResources#read-only-properties-from-get-targetresource-4

---

*[README.md, line 47 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1vcsHmHl__tDlixR_:-Ko1vcsHmHl__tDlixRa:bf1glxr) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L47)):*
> ```Markdown
>  a Hyper-V virtual machine or to the management OS.
> 
> ### xVhd
> ```

Between this and the parameters should we add the resource description? For now we could just copy the description from the ToC? If so, do that for all resources.

---

*[README.md, line 53 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1ynrdmRUPdGmw3Pso:-Ko1ynrdmRUPdGmw3Psp:bss2pdi) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L53)):*
> ```Markdown
> * **`[String]` ParentPath** (_Write_): Parent VHD file path, for differencing disk
> * **`[Uint64]` MaximumSizeBytes** (_Write_): Maximum size of VHD to be created
> * **`[String]` Generation** (_Write_): Virtual disk format: { Vhd | VHDx }
> ```

Add that the default value is 'Vhd'.

---

*[README.md, line 54 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1yLU2STYrYuAKT8nh:-Ko1yLU3zo2PWVgtqkE2:bxholm7) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L54)):*
> ```Markdown
> * **`[Uint64]` MaximumSizeBytes** (_Write_): Maximum size of VHD to be created
> * **`[String]` Generation** (_Write_): Virtual disk format: { Vhd | VHDx }
> * **`[String]` Ensure** (_Write_): Ensures that the VHD is Present or Absent
> ```

Add the value map to this as well. { \*Present*\ | Absent }

Add that the text "Default value is 'Present'."
 (also indicated with \*Present\* in the value map)

---

*[README.md, line 56 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1uyEY7Gd8eWnckWN1:-Ko1uyEY7Gd8eWnckWN2:b1jluqp) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L56)):*
> ```Markdown
> * **`[String]` Ensure** (_Write_): Ensures that the VHD is Present or Absent
> 
> ### xVMHyperV
> ```

This resource has read parameters that should be listed here as well.

---

*[README.md, line 60 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1unZmOjLTVWvVVKRW:-Ko1unZmOjLTVWvVVKRX:b-fg2nwc) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L60)):*
> ```Markdown
> * **`[String]` Name** (_Key_): The desired VM name.
> * **`[String]` VhdPath** (_Required_): The desired VHD associated with the VM.
> * **`[String]` SwitchName** (_Write_): Virtual switch(es) associated with the VM.
> ```

Should be [string[]]

---

*[README.md, line 64 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1zOuh6lmGY30b4yd9:-Ko1zOuiD71apZUGAf0U:bb6ul79) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L64)):*
> ```Markdown
> * **`[String]` State** (_Write_): State of the VM: { Running | Paused | Off }.
> * **`[String]` Path** (_Write_): Folder where the VM data will be stored.
> * **`[Uint32]` Generation** (_Write_): Virtual machine generation { 1 | 2 }.
> ```

Add that the default value is '1'.

---

*[README.md, line 66 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1zj1_CHj1zuuyn1zo:-Ko1zj1_CHj1zuuyn1zp:brtmfy8) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L66)):*
> ```Markdown
> * **`[Uint32]` Generation** (_Write_): Virtual machine generation { 1 | 2 }.
>   Generation 2 virtual machines __only__ support VHDX files.
> * **`[Boolean]` SecureBoot** (_Write_): Enables or disables secure boot
> ```

Add that the default value is $true.

---

*[README.md, line 81 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1yVDvVATnz5vbctKp:-Ko1yVDvVATnz5vbctKq:b-nx7b2z) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L81)):*
> ```Markdown
> * **`[Boolean]` RestartIfNeeded** (_Write_): If specified, will shutdown and
>  restart the VM as needed for property changes.
> * **`[String]` Ensure** (_Write_): Ensures that the VM is Present or Absent.
> ```

Add the value map to this as well. { \*Present*\ | Absent }

Add that the default value is 'Present'.

---

*[README.md, line 83 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1zptp54q2au-bc6XN:-Ko1zptp54q2au-bc6XO:b38ug0b) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L83)):*
> ```Markdown
> * **`[String]` Ensure** (_Write_): Ensures that the VM is Present or Absent.
> * **`[String]` Notes** (_Write_): Notes about the VM.
> * **`[Boolean]` EnableGuestService** (_Write_): Enable Guest Service Interface
> ```

Add that the default value is $false.

---

*[README.md, line 86 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1v8HXj36tmi5AcYo2:-Ko1v8HYgVrgkbY1YYTq:b-ay1hjg) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L86)):*
> ```Markdown
>  for the VM.
> 
> The following xVMHyper-V properties **cannot** be changed after VM creation:
> ```

Maybe move these up under the resource description? So this is shown before the parameters?

---

*[README.md, line 92 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1w6pV1carBail9HsV:-Ko1w6pV1carBail9HsW:b1jluqp) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L92)):*
> ```Markdown
> * Generation
> 
> ### xVMSwitch
> ```

This resource has read parameters that should be listed here as well.

---

*[README.md, line 99 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1zz8PZxT-SYOIWJhj:-Ko1zz8PZxT-SYOIWJhk:b38ug0b) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L99)):*
> ```Markdown
> * **`[String[]]` NetAdapterName** (_Write_): Network adapter name(s)
>  for external switch type
> * **`[Boolean]` AllowManagementOS** (_Write_): Specify if the VM host
> ```

Add that the default value is $false.

---

*[README.md, line 101 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko2-1MSoQi_liAkH4zw:-Ko2-1MSoQi_liAkH4zx:b38ug0b) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L101)):*
> ```Markdown
> * **`[Boolean]` AllowManagementOS** (_Write_): Specify if the VM host
>  has access to the physical NIC
> * **`[Boolean]` EnableEmbeddedTeaming** (_Write_): Should embedded NIC teaming
> ```

Add that the default value is $false.

---

*[README.md, line 103 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko2-4QjffM2cbo1ThMM:-Ko2-4QjffM2cbo1ThMN:b2uigs2) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L103)):*
> ```Markdown
> * **`[Boolean]` EnableEmbeddedTeaming** (_Write_): Should embedded NIC teaming
>  be used (Windows Server 2016 only)
> * **`[String]` BandwidthReservationMode** (_Write_): Specify the QoS mode used
> ```

Add that the default value is 'NA'.

Also make NA italic in the value map.

---

*[README.md, line 106 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1yXLnxSUzctGZgW3E:-Ko1yXLnxSUzctGZgW3F:b-nx7b2z) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L106)):*
> ```Markdown
>  (options other than NA are only supported on Hyper-V 2012+):
>  { Default | Weight | Absolute | None | NA }.
> * **`[String]` Ensure** (_Write_): Ensures that the VM Switch is Present or Absent
> ```

Add the value map to this as well. { \*Present*\ | Absent }

Add that the default value is 'Present'.

---

*[README.md, line 108 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1szeppwH9fESILrAZ:-Ko1szeppwH9fESILrA_:bexk9ql) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L108)):*
> ```Markdown
> * **`[String]` Ensure** (_Write_): Ensures that the VM Switch is Present or Absent
> 
> ### xVhdFile
> ```

Could you order these resources in alphabetical order?

---

*[README.md, line 108 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1xz_Ge9YLFUtQtJza:-Ko1x2OSceI7OkCM3vNq:b-thfixm) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L108)):*
> ```Markdown
> * **`[String]` Ensure** (_Write_): Ensures that the VM Switch is Present or Absent
> 
> ### xVhdFile
> ```

Why is it a class thatis named xFileDirectory in this schema?
https://github.com/bgelens/xHyper-V/blob/dev/DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.schema.mof

---

*[README.md, line 108 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1y0H1qR2s6CeQD7ic:-Ko1y0H1qR2s6CeQD7id:b-podqdi) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L108)):*
> ```Markdown
> * **`[String]` Ensure** (_Write_): Ensures that the VM Switch is Present or Absent
> 
> ### xVhdFile
> ```

Missing parameter CheckSum.

Add default value is 'ModifiedDate' for parameter CheckSum.

---

*[README.md, line 111 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1xs5VDZyfQ5lP4SZ6:-Ko1xs5VDZyfQ5lP4SZ7:bhq67ch) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L111)):*
> [MSFT_xFileDirectory]

Should be `[MSFT_xFileDirectory[]]`

---

*[README.md, line 126 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko2-x2vmqJ_v9OwJ4Pi:-Ko2-x2vmqJ_v9OwJ4Pj:b-fzxcf9) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L126)):*
> ```Markdown
>  file or physical hard disk volume for the added DVD drive.
> * **`[String]` Ensure** (_Write_): Specifies if the DVD Drive should exist or not.
>  { *Present* | Absent }. Defaults to Present.
> ```

Depending on how your write the others this might need to be revised.

---

*[README.md, line 128 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko20G7dtwaUa7qUCrLu:-Ko20G7dtwaUa7qUCrLv:b1jluqp) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L128)):*
> ```Markdown
>  { *Present* | Absent }. Defaults to Present.
> 
> ### xVMNetworkAdapter
> ```

This resource has read parameters that should be listed here as well.

---

*[README.md, line 139 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1y_zsJUzbgNXZlHsI:-Ko1y_zsJUzbgNXZlHsJ:b-lp5lpu) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L139)):*
> ```Markdown
> * **`[String]` MacAddress** (_Write_): Use this to specify a Static MAC Address.
>  If this parameter is not specified, dynamic MAC Address will be set.
> * **`[String]` Ensure** (_Write_): Ensures that the VM Network Adapter is
> ```

Add the value map to this as well. { \*Present*\ | Absent }

And add that the default value is 'Present'.

---

*[README.md, line 386 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko21bBlYd6xl2hgbs1A:-Ko21bBmbnB1FlH-HKOF:b-c4tu55) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L386)):*
> ```Markdown
>     xVMSwitch switch
>     {
>         Name = "Test-Switch"
> ```

Use single quotes on all strings with only text. Do this throughput these examples.

---

*[README.md, line 401 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko21lZUAxCJnYYmXRjq:-Ko21lZUAxCJnYYmXRjr:b-z8ojjl) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L401)):*
> ```Markdown
>     }
> 
>     # Customize VHD by copying a folders/files to the VHD before a VM can be created
> ```

Comment block for this text.

---

*[README.md, line 412 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko228ISG1_RTeEzGfKD:-Ko228ISG1_RTeEzGfKE:b-whl9it) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L412)):*
> # create the generation 1 testVM out of the vhd.

Upper 'C' on Create.

Should be upper-case VHD?

---

*[README.md, line 1151 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko22j_iipHE6gdnZ43u:-Ko22j_iipHE6gdnZ43v:b-tx6kmg) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L1151)):*
> ```Markdown
>         [Parameter(Mandatory=$true, Position=0)]
>         [ValidateScript({Test-Path -Path $_})]
>         [string]
> ```

Maybe use [System.String]? If so, do this throughout.

---

*[README.md, line 1195 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko22vDQ-ccBtwtnibb7:-Ko22vDQ-ccBtwtnibb8:bg2pa0l) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L1195)):*
> ```Markdown
>         [Parameter()]
>         [ValidateSet ("Archive", "Hidden",  "ReadOnly", "System" )]
>         $Attribute
> ```

No type on this one.

---

*[README.md, line 1229 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko23fGqmprU4hSPKwWt:-Ko23dOT-dFJ_lQDcWIK:bm0fxc8) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L1229)):*
> ```Markdown
>     )
> 
>     Import-DscResource -Module xHyper-V
> ```

Are you sure it shouldn't be ModuleName?

https://msdn.microsoft.com/en-us/powershell/dsc/sxsresource

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xhyper-v/113)*
<!-- Sent from Reviewable.io -->

bgelens commented:
@johlju Based on your comments I decided to do the entire overhaul based on how DscResources Readme is done. 

Also added a change log and removed all example code from the Readme and replaced with links.

The example files are the same as the example code was in the Readme. A couple example files where missing, these are created from the code which was in the Readme. 

Examples will be checked and if needed improved (new issue #114 )

---

Review status: 1 of 7 files reviewed at latest revision, 32 unresolved discussions.

---

*[README.md, line 4 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1rGjRMqhBpfWbBCsq:-Ko3oa9CsXWo4X_R2R6P:bjkckh3) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L4)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Do we need to list each resource here? Seems it could be missed when new resources are added. Could we just say something like "using Desired State Configuration." instead?
Or xSQLServer has the text 'The xSQLServer module contains DSC resources for deployment and configuration of SQL Server.'. So maybe something along those lines?

But if you want to keep it as-is, then that looks okay to me.
</blockquote></details>

OK, I thought so myself when adding 2 or 3 which where missing. Changed it

---

*[README.md, line 31 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1sneoRY9-xOxeW-VC:-Ko6GRZKvsyUK3iMiIyE:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L31)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Could you order these resources in alphabetical order?
</blockquote></details>

Done.

---

*[README.md, line 31 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1tHRd5k3qEOOsVZtP:-Ko6GSREeerNiTxEkMa1:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L31)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

The resource and the resource folder is named xVHD (upper 'VHD'). This should reflect that. Or the resource should change the name to 'xVhd'.
</blockquote></details>

Done.

---

*[README.md, line 47 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1tgOSNlQt_erSvyTR:-Ko6GTJWfb6ox4QsqTY5:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L47)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

This resource has read parameters that should be listed here as well.

Example:
https://github.com/PowerShell/PSDscResources#read-only-properties-from-get-targetresource-4
</blockquote></details>

Done.

---

*[README.md, line 47 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1vcsHmHl__tDlixR_:-Ko6GU2dIHGwjmw24ivy:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L47)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Between this and the parameters should we add the resource description? For now we could just copy the description from the ToC? If so, do that for all resources.
</blockquote></details>

Done.

---

*[README.md, line 53 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1ynrdmRUPdGmw3Pso:-Ko6GUt7S_THMVd8uHou:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L53)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Add that the default value is 'Vhd'.
</blockquote></details>

Done.

---

*[README.md, line 54 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1yLU2STYrYuAKT8nh:-Ko6GVuHMpzIl7ACmvZv:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L54)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Add the value map to this as well. { \*Present*\ | Absent }

Add that the text "Default value is 'Present'."
 (also indicated with \*Present\* in the value map)
</blockquote></details>

Done.

---

*[README.md, line 56 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1uyEY7Gd8eWnckWN1:-Ko6hkVMdfNWh_FzhLeZ:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L56)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

This resource has read parameters that should be listed here as well.
</blockquote></details>

Done.

---

*[README.md, line 60 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1unZmOjLTVWvVVKRW:-Ko6hvol7iCN0W-LbiYh:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L60)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Should be [string[]]
</blockquote></details>

Done.

---

*[README.md, line 64 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1zOuh6lmGY30b4yd9:-Ko6iJeYzspUr8ssEwRj:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L64)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Add that the default value is '1'.
</blockquote></details>

Done.

---

*[README.md, line 66 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1zj1_CHj1zuuyn1zo:-Ko6iVeTuNLrKZSKCrp0:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L66)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Add that the default value is $true.
</blockquote></details>

Done.

---

*[README.md, line 81 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1yVDvVATnz5vbctKp:-Ko6iyBrXVx4OwiLr54v:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L81)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Add the value map to this as well. { \*Present*\ | Absent }

Add that the default value is 'Present'.
</blockquote></details>

Done.

---

*[README.md, line 83 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1zptp54q2au-bc6XN:-Ko6j3EXuqBbyqXhtwD-:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L83)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Add that the default value is $false.
</blockquote></details>

Done.

---

*[README.md, line 86 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1v8HXj36tmi5AcYo2:-Ko6jAuFU47mUkMQcUkC:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L86)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Maybe move these up under the resource description? So this is shown before the parameters?
</blockquote></details>

Done.

---

*[README.md, line 92 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1w6pV1carBail9HsV:-Ko6rb0r0jGCGfY4HYtm:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L92)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

This resource has read parameters that should be listed here as well.
</blockquote></details>

Done.

---

*[README.md, line 99 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1zz8PZxT-SYOIWJhj:-Ko6rbZXNPsbOOm-cLI7:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L99)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Add that the default value is $false.
</blockquote></details>

Done.

---

*[README.md, line 101 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko2-1MSoQi_liAkH4zw:-Ko6rfUVpDPPgVcHI0lX:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L101)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Add that the default value is $false.
</blockquote></details>

Done.

---

*[README.md, line 103 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko2-4QjffM2cbo1ThMM:-Ko6rlxgKYgJTNKFTOtt:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L103)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Add that the default value is 'NA'.

Also make NA italic in the value map.
</blockquote></details>

Done.

---

*[README.md, line 106 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1yXLnxSUzctGZgW3E:-Ko6roAtoDQfyDCHqF75:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L106)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Add the value map to this as well. { \*Present*\ | Absent }

Add that the default value is 'Present'.
</blockquote></details>

Done.

---

*[README.md, line 108 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1szeppwH9fESILrAZ:-Ko6GGeZLhNYeqRZ4SIF:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L108)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Could you order these resources in alphabetical order?
</blockquote></details>

Done.

---

*[README.md, line 108 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1xz_Ge9YLFUtQtJza:-Ko6CbrjN6NKHpqFW-qw:btfnmcg) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L108)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Why is it a class thatis named xFileDirectory in this schema?
https://github.com/bgelens/xHyper-V/blob/dev/DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.schema.mof
</blockquote></details>

I think this is done for Intellisense purposes and modeled after the File resource [https://msdn.microsoft.com/en-us/powershell/dsc/fileresource](https://msdn.microsoft.com/en-us/powershell/dsc/fileresource)

I'll add a definition of the class to the readme so people will know the properties. I'll copy in descriptions coming from the above link where applicable. (I won't say what values are default or not as from looking at the code this doesn't looks like the implementation reflects the documentation from the link above.

I won't judge the validity of this approach. I'll leave that to redesign work for the HQRM module.

---

*[README.md, line 108 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1y0H1qR2s6CeQD7ic:-Ko6GMxxZ0IRRO21SVNt:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L108)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Missing parameter CheckSum.

Add default value is 'ModifiedDate' for parameter CheckSum.
</blockquote></details>

Done.

---

*[README.md, line 111 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1xs5VDZyfQ5lP4SZ6:-Ko6GNbcGLpM0uKocb5P:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L111)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> [MSFT_xFileDirectory]

Should be `[MSFT_xFileDirectory[]]`
</blockquote></details>

Done.

---

*[README.md, line 126 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko2-x2vmqJ_v9OwJ4Pi:-Ko6VZv_rEnaPK6u9yYB:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L126)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Depending on how your write the others this might need to be revised.
</blockquote></details>

Done.

---

*[README.md, line 128 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko20G7dtwaUa7qUCrLu:-Ko6lPRVdFRcJEXn_kJ4:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L128)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

This resource has read parameters that should be listed here as well.
</blockquote></details>

Done.

---

*[README.md, line 139 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko1y_zsJUzbgNXZlHsI:-Ko6lREOoJPOQU8Xn_vr:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L139)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Add the value map to this as well. { \*Present*\ | Absent }

And add that the default value is 'Present'.
</blockquote></details>

Done.

---

*[README.md, line 386 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko21bBlYd6xl2hgbs1A:-Ko7K3zIpDn6L2qoSECq:bxor7in) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L386)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Use single quotes on all strings with only text. Do this throughput these examples.
</blockquote></details>

OK. Instead of fixing the example code, I decided to remove it from the readme and link to the examples in the examples folder instead and untouched. Examples where duplicates with a few exceptions. These exceptions are now example files.

---

*[README.md, line 401 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko21lZUAxCJnYYmXRjq:-Ko7K3BbvR0SG_2jRMvF:bxor7in) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L401)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Comment block for this text.
</blockquote></details>

OK. Instead of fixing the example code, I decided to remove it from the readme and link to the examples in the examples folder instead and untouched. Examples where duplicates with a few exceptions. These exceptions are now example files.

---

*[README.md, line 412 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko228ISG1_RTeEzGfKD:-Ko7K1YHfs49trqZQ4pK:bxor7in) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L412)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> # create the generation 1 testVM out of the vhd.

Upper 'C' on Create.

Should be upper-case VHD?
</blockquote></details>

OK. Instead of fixing the example code, I decided to remove it from the readme and link to the examples in the examples folder instead and untouched. Examples where duplicates with a few exceptions. These exceptions are now example files.

---

*[README.md, line 1151 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko22j_iipHE6gdnZ43u:-Ko7JxcA-s-Zjtv_T9WR:bxor7in) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L1151)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Maybe use [System.String]? If so, do this throughout.
</blockquote></details>

OK. Instead of fixing the example code, I decided to remove it from the readme and link to the examples in the examples folder instead and untouched. Examples where duplicates with a few exceptions. These exceptions are now example files.

---

*[README.md, line 1195 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko22vDQ-ccBtwtnibb7:-Ko7JwfystNiKZ_jv5Nd:bxor7in) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L1195)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

No type on this one.
</blockquote></details>

OK. Instead of fixing the example code, I decided to remove it from the readme and link to the examples in the examples folder instead and untouched. Examples where duplicates with a few exceptions. These exceptions are now example files.

---

*[README.md, line 1229 at r1](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-Ko23fGqmprU4hSPKwWt:-Ko7JYjBup-P22CYU-kg:bxor7in) ([raw file](https://github.com/powershell/xhyper-v/blob/ec5edf79fad7768bf9438e7bd35017b4f5e5a001/README.md#L1229)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Are you sure it shouldn't be ModuleName?

https://msdn.microsoft.com/en-us/powershell/dsc/sxsresource
</blockquote></details>

OK. Instead of fixing the example code, I decided to remove it from the readme and link to the examples in the examples folder instead and untouched. Examples where duplicates with a few exceptions. These exceptions are now example files.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-:-Ko7LRrDe-aUM4khivpK:b-t09a4v)*
<!-- Sent from Reviewable.io -->

johlju commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 6 of 6 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xhyper-v/113#-:-Ko7QljKBq8AMThsdgyS:bnfp4nl)*
<!-- Sent from Reviewable.io -->
